src/components/pages/reports/ReportsContainer.js

```diff
@@ -1,83 +1,100 @@
 
-  const filterBy = (field, answers, value) => {
-    switch (field) {
-      case 'aoi':
-        return answers.filter((answer) => answer.aoi === value);
-      case 'date':
-        return answers.filter((answer) => answer.date === value);
-      case 'search':
-        return answers.filter((answer) => {
-          return Object.keys(answer).some((key) => {
-            if(answer[key].toLowerCase().includes(value.toLowerCase())) return true;
-            return false;
-          })
-        })
-      default:
-        break;
+import { DEFAULT_FORMAT, DEFAULT_LANGUAGE } from '../../../constants/global';
+import qs from 'query-string';
+import Reports from './Reports';
+import { filterBy } from '../../../helpers/filters';
+
+
+const getAnswersByTemplate = (templateId, reports) => {
+  const reportIds = reports.answers[templateId].ids;
+  const reportData = reports.answers[templateId].data;
+  let answers = reportIds.map((reportId) => ({
+    id: reportData[reportId].id,
+    date: moment(reportData[reportId].attributes.createdAt).format(DEFAULT_FORMAT),
+    latLong: reportData[reportId].attributes.userPosition.toString(),
+    member: reportData[reportId].attributes.user,
+    aoi: reportData[reportId].attributes.areaOfInterest || null
+  }));
+  return answers;
+}
+
+const getAnswersAreas = (answers, areas) => {
+  const areasIndex = [];
+  const areasOptions = [];
+  answers.forEach((answer) => {
+    if (areas.ids.indexOf(answer.aoi) > -1 && areasIndex.indexOf(answer.aoi) === -1) {
+      areasIndex.push(answer.aoi);
+      areasOptions.push({ label: areas.data[answer.aoi] && areas.data[answer.aoi].attributes.name, value: answer.aoi });
     }
-    return answers.filter((answer) => answer.aoi === value);
-  }
+  });
+  return areasOptions;
+}
 
-  const mapStateToProps = ({ reports, templates }, { match, location }) => {
-    const templateId = templates.ids[match.params.templateIndex || 0];
-    const searchParams = qs.parse(location.search);
-    const { aoi, date, searchValues } = searchParams;
-    let answers = [];
-    if (templateId !== undefined){
-      const selectedAnswers = reports.answers[templateId];
-      if (selectedAnswers !== undefined) {
-        answers = selectedAnswers.map((answer) => ({
-          id: answer.id,
-          date: moment(answer.attributes.createdAt).format(DEFAULT_FORMAT),
-          latLong: answer.attributes.userPosition.toString(),
-          member: answer.attributes.user,
-          aoi: answer.attributes.layer // TODO: Change to AOI instead of layer
-        }))
-        if (aoi !== undefined){ 
-          answers = filterBy('aoi', answers, aoi);
-        }
-        if (searchValues !== undefined){ 
-          answers = filterBy('search', answers, searchValues);
-        }
-        if (date !== undefined){ 
-          answers = filterBy('date', answers, date);
-        }
-      }
+const getTemplateOptions = (templates) => {
```

Comment:
Why don't do that on the `ReportsFiltersContainer` directly?
I mean we are doing in this container and then pass as props to the ReportsFilter component.
Also let you know that the filter is not being populate correctly 
![image](https://user-images.githubusercontent.com/10500650/27955197-f7dd3b4e-6313-11e7-9143-9573cb981f53.png)